sympy/geometry/parabola.py

```diff
@@ -406,3 +407,21 @@ def vertex(self):
             vertex = Point(focus.args[0], focus.args[1] - self.p_parameter)
 
         return vertex
+from sympy import *
+def tangent(a,b,g,f,h,c,x1,y1):  
+    x,y=symbols('x y')
+    P=Poly((x**2)*a+(y**2)*b+2*g*x+2*h*x*y+2*f*y+c,(x,y))
+    S=Eq((x**2)*a+(y**2)*b+(2*g*x)+(2*h*x*y)+(2*f*y),-c)
+    S1=P(x1,y1)
+    if S1==0:
+        print('Point lies on the curve')
+    elif S1<0:
+        print('Point lies inside the curve')
+    else:
+        print('Point lies outside the curve')
```

Comment:
You don't need to print where the point lies. 

If someone calls the tangent function for a parabola, he's only concerned with the equation of the tangent to the parabola passing through that point,  not where the point lies.

Reply:
Also, you should throw a `ValueError` for a point inside the curve as it cannot have a tangent passing through it